doc/techsupport-manager/core_file_manager.md

```diff
@@ -0,0 +1,329 @@
+ - A SONiC service called core file manager is added to provide additional management and export of the core dumps that systemd-coredump generates to a remote host.
+ - The core file manager service also includes "show tech-support" data along with the core file when exporting crash reports to remote host. Exporting of crash reports is done both periodically and on discovery of a new core file.
+ - SONiC UI capabilities are added to manage the core file manager service and systemd-coredump functionality.
+This document describes the high level design details of core file manager framework that consists of core file and tech-support data generation framework. It also provides the exporting of  tech-support data to a remote server when a new core is generated and also on a periodic interval.  This document describes the tools used to support the core file generation and export functionality.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+
+| **Term**     |  **Meaning**                  |
+|:-------------|:------------------------------|
+| tech-support | Technical support data        |
+| SCP          | Secure Copy Protocol          |
+| SFTP         | Secure File Transfer Protocol |
+
+
+## Requirements Overview
+
+### Functional Requirements
+
+This document describes new mechanisms to manage the core files that are generated in a SONiC environment. When a process is terminated unexpectedly, the System generates the core dump of that process in most cases. The core files are stored on the persistent storage like SSDs. These core files need to be managed on the SONiC devices to optimize disk space usage, exported to remote server for offline analysis and recorded to maintain a history of occurrences. In addition, existing "tech-support" information is associated with the core dumps and managed similarly.
+
+### Core Manager Requirements
+
+  1. Add a new SONiC service to manage the core dumps as follows: - 
+
+	 a.  Support per-process core file rotation and archiving to optimize disk space 
+
+ 	 b.  Strip Core files sensitive information 
+
+ 2.  Add a new SONIC service to collect and export the tech-support data as follows:-
+
+	 a.  Collect tech-support data when a new core dump is discovered, and export it to a remote server.
+
+	 b. Periodically collect and export tech-support data
+
+
+### Configuration and Management Requirements
+
+To configure the core dump and tech-support data, export to an external server and to view the core details the following config and show commands shall be supported. It is to be noted that the tech-support data always includes the core dumps generated on the system.
+
+### Config commands
+
+>1. Config command to enable/disable the coredump generation of processes.
+>2.  Config command to store the details of exporting tech-support data to an external server which includes remote server name, path, transfer protocol type and the user credentials.
```

Comment:
Can we register an external script to be invoked for uploading ?

This way we can extend the ability to store anywhere. For example, azure-cloud-storage

Reply:
No external script. It uses the standard protocol scp/sftp to update the techsupport data to remote server.

Reply:
Sure. But I think, accepting a script (_with full path_) as part of remote server configuration (_that takes server address, user, pass,..._) will make this transparently extendable. You already have the FW, so adding this can be simple, I believe.

Reply:
Side note: 
You don't imply "coredump *generation* of processes", but imply "coredump upload", correct.